src/snowflake/mod.rs

```diff
@@ -301,10 +401,69 @@ pub(crate) async fn build_store_for_snowflake_stage(
                 client
             });
 
-            Ok((Arc::new(store), crypto_material_provider, stage_prefix.to_string(), extension))
+            Ok((Arc::new(store), crypto_material_provider, Some(stage_prefix.to_string()), extension))
+        }
+        "AZURE" => {
+            let (container, stage_prefix) = info.stage_info.location.split_once('/')
+                .ok_or_else(|| Error::invalid_response("Stage information from snowflake is missing the container name"))?;
+            let storage_account = info.stage_info.storage_account
+                .clone()
+                .ok_or_else(|| Error::invalid_response("Stage information from snowflake is missing the storage account name"))?;
+
+
+            let provider = AzureStageCredentialProvider::new(&config.stage, client.clone());
+
+            match info.stage_info.test_endpoint.as_deref() {
+                Some(_) => {
+                    unimplemented!("test endpoint for azure blob storage is not supported");
+                }
+                None => {}
+            }
```

Comment:
That could be written like this (because you don't need the value):
```suggestion
            if info.stage_info.test_endpoint.is_some() {
                unimplemented!("test endpoint for azure blob storage is not supported");
            }
```

Reply:
uuhhh an `if` in Rust :D A rare sight!  Done. Thanks for the suggestion!